File: C++/1051. HeightChecker.cpp
//Return the minimum number of students that must move in order for all students to be standing in non-decreasing order of height.
//Make a copy of the array, sort the copy and comapre it with the actual array

class Solution {
public:
        int heightChecker(vector<int>& heights) {
        if (heights.size()<2)
            return 0;
        
        vector<int> map(101,0);
        
        for (int h:heights)
            map[h]++;
        
        int res = 0;
        int h_ptr = 1;
        
        for (int h:heights)
        {
            while (map[h_ptr]==0)
                h_ptr++;
            
            if (h != h_ptr)
                res++;
            
            map[h_ptr]--;
        }
        
        return res;
    }
};

File: Java/1051. HeightChecker.java
import java.util.*;
class Solution {
    public int heightChecker(int[] heights) {
        int[] sortHeights=new int[heights.length];
        int count=0;
        for(int i=0;i<heights.length;i++){
            sortHeights[i]=heights[i];
        }
        Arrays.sort(sortHeights);
        for(int i=0;i<heights.length;i++){
            
            if(heights[i]!=sortHeights[i]){
                count++;
            }
        }
        return count;
    }
}
